Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -16,44 +16,41 @@
 
 #include "Starboard.h"
 #import <StubReturn.h>
+#include <vector>
 
-#include "Foundation/NSOrderedSet.h"
+#import <Foundation/NSOrderedSet.h>
+#import <Foundation/NSException.h>
+
+@implementation NSOrderedSet {
+    std::vector<StrongId<id>> setContainer;
```

Comment:
I can switch the backing storage to a dictionary.
but then only the value gets retained in a dictionary. thus the contains would still be O(N).
or I can use an map have the key wrapped  in StrongId, but still then it would be (obj,1), so objectAtIndex would not be O(1)